utopia-vscode-extension/src/extension.ts

```diff
@@ -2,60 +2,56 @@ import * as vscode from 'vscode'
 import type {
   DecorationRange,
   DecorationRangeType,
-  FSError,
   BoundsInFile,
   Bounds,
-  ToVSCodeMessage,
   UtopiaVSCodeConfig,
+  FromUtopiaToVSCodeMessage,
+  FromVSCodeToUtopiaMessage,
 } from 'utopia-vscode-common'
 import {
-  ensureDirectoryExists,
-  RootDir,
-  initMailbox,
-  VSCodeInbox,
-  setErrorHandler,
   toUtopiaPath,
-  initializeFS,
-  parseToVSCodeMessage,
-  sendMessage,
   editorCursorPositionChanged,
-  readFileAsUTF8,
-  exists,
-  writeFileUnsavedContentAsUTF8,
-  clearFileUnsavedContent,
   applyPrettier,
   utopiaVSCodeConfigValues,
   vsCodeReady,
   clearLoadingScreen,
   ProjectIDPlaceholderPrefix,
+  vsCodeBridgeReady,
+  vsCodeFileChange,
 } from 'utopia-vscode-common'
 import { UtopiaFSExtension } from './utopia-fs'
-import { fromUtopiaURI } from './path-utils'
 import type { TextDocumentChangeEvent, TextDocumentWillSaveEvent, Uri } from 'vscode'
+import type { FSError } from './in-mem-fs'
+import {
+  clearFileUnsavedContent,
+  exists,
+  readFileAsUTF8,
+  setErrorHandler,
+  writeFileUnsavedContentAsUTF8,
+} from './in-mem-fs'
 
 const FollowSelectionConfigKey = 'utopia.editor.followSelection.enabled'
 
-export async function activate(context: vscode.ExtensionContext): Promise<void> {
+export function activate(context: vscode.ExtensionContext) {
```

Comment:
Since we're no longer relying on IndexedDB, all of the FS functionality is now sync